src/data/ability.ts

```diff
@@ -5027,22 +5028,21 @@ export class PostDamageForceSwitchAbAttr extends PostDamageAbAttr {
 
     // Dragon Tail and Circle Throw switch out Pokémon before the Ability activates.
     const fordbiddenDefendingMoves = [ Moves.DRAGON_TAIL, Moves.CIRCLE_THROW ];
-    const getField = [ ...pokemon.getOpponents(), ...pokemon.getAlliedField() ];
-    for (const opponent of getField) {
-      const enemyMoveHistory = opponent.getMoveHistory();
+    if (source) {
+      const enemyMoveHistory = source.getMoveHistory();
       if (enemyMoveHistory.length > 0) {
         const enemyLastMoveUsed = enemyMoveHistory[enemyMoveHistory.length - 1];
         if (fordbiddenDefendingMoves.includes(enemyLastMoveUsed.move) || enemyLastMoveUsed.move === Moves.SKY_DROP && enemyLastMoveUsed.result === MoveResult.OTHER) {
           return false;
         // Will not activate if the Pokémon's HP falls below half by a move affected by Sheer Force.
-        } else if (allMoves[enemyLastMoveUsed.move].chance >= 0 && opponent.hasAbility(Abilities.SHEER_FORCE)) {
+        } else if (allMoves[enemyLastMoveUsed.move].chance >= 0 && source.hasAbility(Abilities.SHEER_FORCE)) {
           return false;
         // Activate only after the last hit of multistrike moves
-        } else if (opponent.turnData.hitsLeft > 1) {
+        } else if (source.turnData.hitsLeft > 1) {
           return false;
         }
-        const multiHitModifier = opponent.getHeldItems().find(m => m instanceof PokemonMultiHitModifier);
-        if (allMoves[enemyLastMoveUsed.move].hasAttr(MultiHitAttr) || multiHitModifier) {
+        const multiHitModifier = source.getHeldItems().find(m => m instanceof PokemonMultiHitModifier);
+        if (allMoves[enemyLastMoveUsed.move].hasAttr(MultiHitAttr) || multiHitModifier || source.hasAbilityWithAttr(AddSecondStrikeAbAttr)) {
```

Comment:
```suggestion
        if (source.turnData.hitCount > 1) {
```
This should cover any case where an attack strikes more than once